rslib/src/scheduler/fsrs/memory_state.rs

```diff
@@ -307,11 +308,15 @@ pub(crate) fn single_card_revlog_to_item(
             }))
         } else if let Some(first_review) = first_review {
             // the revlog has been truncated, but not fully
-            let starting_state = fsrs.memory_state_from_sm2(
+            let mut starting_state = fsrs.memory_state_from_sm2(
                 first_review.ease_factor,
                 first_review.interval,
                 historical_retention,
             )?;
+            // if the ease factor is less than 1.1, the revlog entry is generated by FSRS
+            if first_review.ease_factor <= 1.1 {
+                starting_state.difficulty = (first_review.ease_factor - 0.1) * 9.0 + 1.0;
```

Comment:
Why is this `ease_factor - 0.1`? Isn't the formula `D = (ease factor) * 9 + 1`?

Reply:
```
    /// Returns the difficulty normalized to a 0.1-1.1 range,
    /// which is used in revlog entries.
    pub(crate) fn difficulty_shifted(&self) -> f32 {
        self.difficulty() + 0.1
    }
```

Reply:
So, the difficulty is normalized twice, first from 1-10 to 0-1 and then to 0.1-1.1. Interesting!